keel-core/src/main/kotlin/com/netflix/spinnaker/keel/persistence/UnhappyVetoRepository.kt

```diff
@@ -67,9 +67,9 @@ abstract class UnhappyVetoRepository(
    */
   abstract fun getAllForApp(application: String): Set<String>
 
-  fun calculateExpirationTime(wait: Duration): Instant =
+  fun calculateExpirationTime(wait: Duration?): Instant? =
     when (wait) {
-      Duration.ZERO -> Instant.ofEpochMilli(Long.MAX_VALUE)
+      null -> null
```

Comment:
Here it is, the first good use of `null` I've seen.

Reply:
I don't think null per se is the problem. It's languages (like Java) that fail to distinguish the possibility of a variable being null so it's very easy to accidentally mishandle potentially null values. That's something Kotlin (mostly) fixes (not that it's the first language to do so).

Reply:
I'm a fan of Haskell's Maybe monad for dealing with this sort of thing.

Reply:
I've made the observation before that Kotlin's nullable types are an option/maybe monad with some syntactic sugar.